yt/frontends/flash/io.py

```diff
@@ -32,6 +32,15 @@ def determine_particle_fields(handle):
     return _particle_fields
 
 
+def _conditionally_split_arrays(inp_arrays, condition):
```

Comment:
I'm confused by the name of this function, it doesn't look to me like it does what it says, could we rename it ?

Reply:
I mean, it does actually?  It splits up the arrays based on where the condition is true.

Reply:
oh my. Reading it I figured it would return mask arrays. More coffee required, carry on !